### PR TITLE
Remove 'default' context and require context prop.

### PR DESCRIPTION
**Pull request by pvh**

### CODE REVIEW

File: src/renderer/components/content.js [hunk 1]
```diff
@@ -12,16 +12,12 @@ const FILTERED_PROPS = ['type', 'hypermergeUrl']
...
-
-  static defaultProps = {
-    context: 'default'
+    context: PropTypes.string.isRequired
   }
 
   static initializeContentDoc = (type, typeAttrs = {}) => {
     const { repo } = window // still not a great idea
-    const contentType = ContentTypes.lookup({ type })
+    const contentType = ContentTypes.lookup({ type, context: 'workspace' })
```
Comment (pvh) [inline src/renderer/components/content.js]: We talked about moving this "up" to the type definition. Let's do that in a separate patch though.

File: src/renderer/components/workspace/workspace.js [hunk 2]
```diff
@@ -211,7 +211,7 @@ ContentTypes.register({
   type: 'workspace',
   name: 'Workspace',
   icon: 'briefcase',
-  contexts: { default: Workspace },
+  contexts: { workspace: Workspace },
```
Comment (pvh) [inline src/renderer/components/workspace/workspace.js]: This is the one thing I think is "wrong", in that the workspace is not in a workspace context. We should probably register it as, say, "root"?

File: src/renderer/content-types.js [hunk 3]
```diff
@@ -39,10 +39,12 @@ function registerDefault(contentType) {
...
 
-function lookup({ type, context = 'workspace' } = {}) {
-  if (registry[type] && registry[type].contexts[context]) {
-    return { ...registry[type], component: registry[type].contexts[context] }
+function lookup({ type, context } = {}) {
+  const component = registry[type] && registry[type].contexts[context]
+  if (component) {
+    return { ...registry[type], component }
   }
+
```
Comment (pvh) [inline src/renderer/content-types.js]: Reading the patch, I think you can probably combine the two cases a little further (default vs non-default) so that there's less risk of subtle bugs being introduced.

File: src/renderer/index.js [hunk 4]
```diff
@@ -103,7 +103,7 @@ function initWorkspace() {
     saveWorkspaceUrl(newWorkspaceUrl)
     workspaceUrl = newWorkspaceUrl
   }
-  const workspace = <Content url={workspaceUrl} />
+  const workspace = <Content context="workspace" url={workspaceUrl} />
```
Comment (pvh) [inline src/renderer/index.js]: (this is the other spot for 'root'.)

File: src/renderer/components/workspace/workspace.js (hunk 2 above)
Comment (jeffpeterson) [reply src/renderer/components/workspace/workspace.js]: Ohhh, the "workspace" context is something rendered as the child of the workspace? Okay, that makes sense.

File: src/renderer/components/content.js (hunk 1 above)
Comment (jeffpeterson) [reply src/renderer/components/content.js]: Yeah exactly

File: src/renderer/content-types.js (hunk 3 above)
Comment (jeffpeterson) [reply src/renderer/content-types.js]: Merged the branches and cleaned it up a bit

File: src/renderer/components/pdf-content.js [hunk 5]
```diff
@@ -57,8 +57,10 @@ export default class PDFCard extends React.PureComponent {
 
   loadPDF = (hyperfileUrl) => {
     Hyperfile.fetch(hyperfileUrl, (hyperfileContents) => {
-      this.setState({ currentHyperfileUrl: hyperfileUrl,
-        reactPDFData: { data: hyperfileContents } })
+      this.setState({
+        currentHyperfileUrl: hyperfileUrl,
+        reactPDFData: { data: hyperfileContents }
+      })
```
Comment (jeffpeterson) [inline src/renderer/components/pdf-content.js]: Whoops, auto-format. I disabled prettier, but it must have fallen pack to the default formatter.
